lib/src/main/java/uk/gov/di/data/lep/library/dto/deathnotification/DeathNotificationSetMapper.java

```diff
@@ -175,12 +175,39 @@ private static String getAliasNameTypeOrNull(List<String> deceasedAliasNameTypes
     private static Name generateName(GroPersonNameStructure nameStructure, String description) {
         var givenNameParts =
             nameStructure.personGivenNames() == null
-                ? Stream.of(new NamePart(NamePartType.GIVEN_NAME, ""))
+                ? null
                 : nameStructure.personGivenNames().stream().map(n -> new NamePart(NamePartType.GIVEN_NAME, n)
             );
-        return new Name(
-            description,
-            Stream.concat(givenNameParts, Stream.of(new NamePart(NamePartType.FAMILY_NAME, nameStructure.personFamilyName()))).toList()
-        );
+
+        if (description == null) {
+            return new Name(
+                null,
+                givenNameParts == null
```

Comment:
This feels a bit repeated to me (multiple checks if given names are null) but I'm not entirely convinced there's a good way around that

Reply:
I think it's probably just me being sad there's no nullish coalescing operator 🥲